src/registerServiceWorker.ts

```diff
@@ -19,8 +19,11 @@ if (process.env.NODE_ENV === 'production') {
     updatefound() {
       console.log('New content is downloading.');
     },
-    updated() {
+    updated(registration) {
       console.log('New content is available; please refresh.');
+      // Just immediately refresh. In the future we can add UI to prompt the user to refresh.
+      registration?.waiting?.postMessage({ type: 'SKIP_WAITING' });
```

Comment:
Why are the question marks needed?

Reply:
@mjp2233 TypeScript knows that `registration` could theoretically be null, and complains if you don't handle it. The question marks are called "optional chaining." I'm pretty sure it will never be null in practice, but if we find that it is null in some conditions we can implement more robust handling depending on what those conditions are and how we want to deal with them.